sktime/transformations/series/window_summarizer.py

```diff
@@ -0,0 +1,334 @@
+#!/usr/bin/env python3 -u
+# -*- coding: utf-8 -*-
+# copyright: sktime developers, BSD-3-Clause License (see LICENSE file)
+"""Extract features across (shifted) windows from provided series."""
+
+__author__ = ["danbartl"]
+__all__ = ["WindowSummarizer"]
+
+import pandas as pd
+from joblib import Parallel, delayed
+
+from sktime.transformations.base import BaseTransformer
+
+
+class WindowSummarizer(BaseTransformer):
+    """Transformer for extracting time series features.
+
+    The WindowSummarizer transforms input series to features
+    based on a provided dictionary of window summarizer, window shifts
+    and window lengths.
+
+    Parameters
+    ----------
+    Dictionary with the following arguments:
+    index : str, base string of the derived features
+            final name will also contain
+            window shift and window length.
+    summarizer: either str corresponding to implemented pandas function, currently
+            * "sum",
+            * "mean",
+            * "median",
+            * "std",
+            * "var",
+            * "kurt",
+            * "min",
+            * "max",
+            * "corr",
+            * "cov",
+            * "skew",
+            * "sem"
+            or function definition
+    window: list of integers
+        Contains values for window shift and window length.
+    target_cols: list of str,
+        Specifies which columns in X to target. If set to None will
+        target first column in X.
+
+    Examples
+    --------
+    >>> import pandas as pd
+    >>> from sktime.transformations.series.window_summarizer import WindowSummarizer
+    >>> from sktime.datasets import load_airline, load_longley
+    >>> from sktime.forecasting.naive import NaiveForecaster
+    >>> from sktime.forecasting.base import ForecastingHorizon
+    >>> from sktime.forecasting.compose import ForecastingPipeline
+    >>> from sktime.forecasting.model_selection import temporal_train_test_split
+    >>> y = load_airline()
+    >>> kwargs = WindowSummarizer.get_test_params()[0]
+    >>> transformer = WindowSummarizer(**kwargs)
+    >>> y_transformed = transformer.fit_transform(y)
+    >>> # Example y univariate
```

Comment:
you can split the example into separate sections, would look like this: https://www.sktime.org/en/latest/api_reference/auto_generated/sktime.forecasting.model_selection.ForecastingGridSearchCV.html?highlight=forecastinggridsearch#sktime.forecasting.model_selection.ForecastingGridSearchCV